### ISSUE TITLE
Be Careful What You Backpropagate: A Case For Linear Output Activations & Gradient Boosting

### ISSUE THREAD

**Issue by icoxfog417** [arXivTimes/arXivTimes#361]
## 一言でいうと

ニューラルネットでは層を積むほど複雑性が増し最適化が難しくなる。なので必要以上に層を積まない方がいいが、この不必要の筆頭が表現力に貢献しないが非線形である出力層(softmaxなど)だ！という指摘。出力は線形、伝搬誤差は累乗(強調)することで精度と収束速度の向上が確認できた。

### 論文リンク

https://arxiv.org/abs/1707.04199

### 著者/所属機関

Anders Oland, Aayush Bansal, Roger B. Dannenberg, Bhiksha Raj

* Carnegie Mellon University

## 結果

MNISTで検証したところ、線形の方が収束速度が高く精度も変わりがないことを確認。

![image](https://user-images.githubusercontent.com/544269/28245291-6c665f82-6a3e-11e7-8f6b-63867c35805f.png)

しかし、CIFARで行ったところ逆の結果となった。これは、softmaxの計算においてexponentialが使われており、これにより誤差があった個所が強調されることが理由と考えられる。そのため、出力は線形のままで伝搬誤差にexpをかける方法を考案。

![image](https://user-images.githubusercontent.com/544269/28245309-9b80cce4-6a3e-11e7-94df-d421419a3ea1.png)

また、符号を考慮するためにexpでなく三乗を利用する方法も提案(pow-3G)。これによっても大きな効果が得られた。

![image](https://user-images.githubusercontent.com/544269/28245300-7f823654-6a3e-11e7-85f0-49364e9ed9ef.png)

**Comment by icoxfog417** [comment]
検証ブログ

http://anishshah.github.io/ml/2017/07/17/Gradient-Boosting.html
